cpp/libcudf_kafka/src/kafka_consumer.cpp

```diff
@@ -150,21 +150,29 @@ int64_t kafka_consumer::get_committed_offset(std::string const &topic, int parti
                "Failed retrieve Kafka committed offsets");
 
   int64_t offset = toppar_list[0]->offset();
-  return offset > 0 ? offset : 0;
+  return offset > 0 ? offset : -1001;
 }
 
-std::map<std::string, std::vector<int32_t>> kafka_consumer::list_topics() const
+std::map<std::string, std::vector<int32_t>> kafka_consumer::list_topics(
+  std::string specific_topic) const
 {
-  auto const md = [&]() {
+  auto const metadata = [&]() {
+    std::string errstr;
+    RdKafka::Topic const *const spec_topic =
```

Comment:
Does `spec_topic` need to be manually deleted? I can't find this info in their docs.

Reply:
Yes it does. I'll do that.